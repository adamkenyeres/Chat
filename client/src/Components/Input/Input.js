import React, { Component } from 'react';
import './Input.css'
class Input extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: "",
        }
    }
    render() {
        const {placeholder, buttonText} = this.props
        return (
            <div className="Input">
                <form onSubmit={e => this.onSubmit(e)}>
                    <input
                        onChange={e => this.onChange(e)}
                        value={this.state.text}
                        type="text"
                        placeholder={placeholder}//"Write message"
                        autoFocus="true" />
                    <button> {buttonText} </button>
                </form>
            </div>
        );
    }
    onChange(e) {
        this.setState({text: e.target.value});
    }
    onSubmit(e) {
        e.preventDefault();
        this.setState({text: ""});
        this.props.onSendMessage(this.state.text);
    }
}

export default Input;
