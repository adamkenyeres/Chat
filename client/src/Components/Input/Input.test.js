import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Input from './Input';


describe('Input', () => {
    it('Should render correctly', function(){
        const onSubmitFn = jest.fn();
        const component = renderer.create(<Input onSendMessage={onSubmitFn} placeholder={"Enter username"} buttonText={"Save"} name="username" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('calls onSubmit prop function when form is submitted', () => {
        const onSubmitFn = jest.fn();
        const wrapper = mount(<Input onSendMessage={onSubmitFn} placeholder={"Enter username"} buttonText={"Save"} name="username"/>);
        const form = wrapper.find('form');
        form.simulate('submit');
        expect(onSubmitFn).toHaveBeenCalledTimes(1);
      });
});
