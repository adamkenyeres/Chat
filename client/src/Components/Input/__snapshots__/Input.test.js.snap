// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render correctly 1`] = `
<div
  className="Input"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Enter username"
      type="text"
      value=""
    />
    <button>
       
      Save
       
    </button>
  </form>
</div>
`;
